src/ui/interface.js

```diff
@@ -228,4 +230,67 @@ module.exports = function Interface(options) {
     $("[rel=popover]").popover()
     return true;
   });
+
+  //Start Handle multiple webcam resolutions
+  function changeResolution(w, h){
+    document.getElementById('image').setAttribute("width",w);
+    document.getElementById('image').setAttribute("height",h);
+  }
+  $('#qvga').click(function(e){
+    changeResolution('100px','100px')
+  });
+  $('#vga').click(function(e){
+    changeResolution('800px','600px')
+  });
+  $('#hd').click(function(e){
+    changeResolution('2400px','1800px')
+  })
+  $('#full-hd').click(function(e){
+    changeResolution('6000px','4000px')
+  });
+  //End Handling of Multiple webcam resolutions
+
+  //Start Canvas Recording and Download
+
+  const canvas = document.getElementById('image');
+  const ctx = canvas.getContext('2d');
+  var x = 0;
+  const stream = canvas.captureStream(); // grab our canvas MediaStream
+  const rec = new MediaRecorder(stream); 
+
+  function exportVid(blob) {
+    const vid = document.createElement('video');
+    vid.src = URL.createObjectURL(blob);
+    vid.controls = true;
+    vid.style.display='none';
+    document.body.appendChild(vid);
+    const a = document.createElement('a');
+    a.style.display = 'none';
+    a.href = vid.src;
+    a.download = 'infragramVideo.mp4';
+    document.body.appendChild(a);
+    a.click();
+    setTimeout(() => {
+      document.body.removeChild(a);
+      URL.revokeObjectURL(vid.src);
+    }, 100);
+  }
+  $('#startRecord').click(function(e){
+    const chunks = [];
+
+    rec.ondataavailable = e => chunks.push(e.data);
+    rec.onstop = e => exportVid(new Blob(chunks, {type: 'video/h264'}));
+    rec.start();
+    document.getElementById('startRecord').style.display='none';
+    document.getElementById('stopRecord').style.display='block';
+  })
+  $('#stopRecord').click(function(e){
+    rec.stop();
+    document.getElementById('stopRecord').style.display='none';
+    document.getElementById('startRecord').style.display='block'; 
+    document.getElementById('downloadButton').style.display='block'; 
+  })
```

Comment:
As a whole, it's functioning well when I test it. My only suggestion would be to consider changing the background color of the record button to red while recording. Perhaps in addition to toggling the display from none/block we can add that here. Great job, @Forchapeatl 🚀

Reply:
**Before** Recording

![Screenshot from 2022-08-17 13-08-38](https://user-images.githubusercontent.com/24577149/185115186-9a0538cd-25af-4c86-bad7-5225a1835aa3.png)

**During** Recording

![Screenshot from 2022-08-17 13-09-10](https://user-images.githubusercontent.com/24577149/185115265-ac810ae4-bb76-4feb-b5b1-886cdb6d56b1.png)

**After** Recording

![Screenshot from 2022-08-17 13-08-38](https://user-images.githubusercontent.com/24577149/185115186-9a0538cd-25af-4c86-bad7-5225a1835aa3.png)